modules/ghlint-snakeyaml/src/test/kotlin/net/twisterrob/ghlint/model/SnakePermissionsTest.kt

```diff
@@ -0,0 +1,112 @@
+package net.twisterrob.ghlint.model
+
+import io.kotest.matchers.shouldBe
+import org.intellij.lang.annotations.Language
+import org.junit.jupiter.api.Test
+
+class SnakePermissionsTest {
+
+	@Test fun `job has permissions with correct values`() {
+		val job = loadJob(
+				"""
+				jobs:
+				  test:
+				    permissions:
+				      contents: read
+				      issues: write
+				    runs-on: ubuntu-latest
+				    steps:
+				      - uses: actions/checkout@v4
+			""".trimIndent()
+		)
+
+		job.permissions?.contents shouldBe Access.READ
+		job.permissions?.issues shouldBe Access.WRITE
+	}
+
+	@Test fun `workflow has permissions with correct values`() {
+		val workflow = loadWorkflow(
+				"""
+				on:
+				  push:
+				permissions:
+				  pull-requests: write
+				  id-token: read
+				jobs:
+				  test:
+				    runs-on: ubuntu-latest
+				    steps:
+				      - uses: actions/checkout@v4
+			""".trimIndent()
+		)
+
+		workflow.permissions?.pullRequests shouldBe Access.WRITE
+		workflow.permissions?.idToken shouldBe Access.READ
+	}
+
+	@Test fun `job with no permissions is null`() {
+		val job = loadJob(
+				"""
+				jobs:
+				  test:
+				    runs-on: ubuntu-latest
+				    steps:
+				      - uses: actions/checkout@v4
+			""".trimIndent()
+		)
+
+		job.permissions shouldBe null
+	}
+
+	@Test fun `job with one permission set, remaining are access NONE`() {
+		val job = loadJob(
+				"""
+				jobs:
+				  test:
+				    permissions:
+				      repository-projects: read
+				    runs-on: ubuntu-latest
+				    steps:
+				      - uses: actions/checkout@v4
+			""".trimIndent()
+		)
+
+		job.permissions?.repositoryProjects shouldBe Access.READ
+
+		Permission.entries.forEach {
+			if (it != Permission.REPOSITORY_PROJECTS) {
+				job.permissions?.accessOf(it) shouldBe Access.NONE
+			}
+		}
```

Comment:
this would scale nicely:
```suggestion
		(Permission.entries - setOf(Permission.REPOSITORY_PROJECTS)).forEach {
			job.permissions?.accessOf(it) shouldBe Access.NONE
		}
```
as a reusable (private) function where setOf can be replaced with `permissions.toSet()` on a `vararg` parameter.

Reply:
still would simplify (and strengthen the first few tests as a side effect too)